package/MDAnalysis/analysis/parallel_jobs.py

```diff
@@ -0,0 +1,244 @@
+            self.threads = mp.cpu_count()
+        elif threads > mp.cpu_count(): 
+            self.threads = mp.cpu_count()
+        else:
+            self.threads = threads
+        
+        self.progressbar = progressbar
+
+
+    def slices(self):
+        # This function returns a list containing the start and
+        # last configuration to be analyzed from each thread
+
+        threads    = self.threads # Get number of threads from initialization
+        step       = self.step
+        configs    = (self.stop-self.start)/step
+
+        # Check whether the number of threads is higher than
+        # the number of trajectories to be analyzed
+        while (configs/threads == 0):
+            threads -= 1
+            
+        self.threads = threads # Update the number of threads
+
+        print "Total configurations: "+str(configs)
+        print "Analysis running on ", threads, " threads.\n"
+
+        # Number of cfgs for each thread, and remainder to be added
+        thread_cfg   = configs/threads
+        reminder   = configs%threads
+
+        slices     = []
+        beg = self.start
+
+        # Compute the start and last configurations
+        for thread in range(0,threads):
+            if thread < reminder:
+                end = beg+step*thread_cfg
+            else:
+                end = beg+step*(thread_cfg-1)
+            
+            slices.append([beg, end+1])
+            beg = end+step
+
+        # Print on screen the configurations assigned to each thread
+        for thread in range(threads):
+            confs = 1+(slices[thread][1]-1-slices[thread][0])/step
+            digits = len(str(self.stop))
+            line = "Thread "+str(thread+1).rjust(len(str(threads)))+": " \
+                          +str(slices[thread][0]).rjust(digits)+"/"  \
+                          +str(slices[thread][1]-1).rjust(digits)    \
+                          +" | Configurations: "+str(confs).rjust(1+len(str(thread_cfg)))
+            print line
+
+        print 
+        return slices
+
+    def job_copies(self):
+        # This function creates n=threads copies of the analysis
+        # objects submitted from the user
+        threads    = self.threads
+        job_list   = []
+
+        for thread in range(threads):
+            job_list.append([ copy.deepcopy(job) for job in self.list_of_jobs])
+            
+        return job_list
+
+    def compute(self,job_list,start,stop,step, out_queue, order,progress,error):
+        # Run the single_frame method for each analysis object for all
+        # the trajectories in the batch
+        count = 0 # just to track IO errors
+
+        try:
+            for i, ts in enumerate(self._trajectory[start:stop:step]):
+                count = i
+                for job in job_list:
+                    job._single_frame()
+                progress.put(order) # Updates the progress bar
+            
+            out_queue.put((job_list,order)) # Returns the results
+        except:
```

Comment:
We're still just looking at this at a design level, so we don't need this level of detail yet.